### ISSUE TITLE
Strange behavior of learning curves

### ISSUE THREAD

**Issue by ShrRa** [JustGlowing/minisom#30]
Hello.
I'm trying to use MiniSOM for my work, but I encountered some strangeness with learning curves.
Simple example: let's take learning curves example from your tutorials and run it with a different number of iterations: 10000, 20000 and 30000.
```from minisom import MiniSom

import numpy as np
from sklearn import preprocessing
import matplotlib.pyplot as plt
from matplotlib.gridspec import GridSpec
import os
import sys

data = np.genfromtxt('iris.csv', delimiter=',', usecols=(0, 1, 2, 3))
# data normalization
scaler = preprocessing.MinMaxScaler()
data = scaler.fit_transform(data)

max_iter = [10000,20000,30000]

for num_iter in max_iter:
    som = MiniSom(7, 7, 4, sigma=1., learning_rate=0.5, 
                  neighborhood_function='gaussian', random_seed=10)
    som.pca_weights_init(data)
    
    q_error_pca_init = []
    iter_x = []
    for i in range(num_iter):
        percent = 100*(i+1)/num_iter
        rand_i = np.random.randint(len(data))
        som.update(data[rand_i], som.winner(data[rand_i]), i, num_iter)
        if (i+1) % 100 == 0:
            error = som.quantization_error(data)
            q_error_pca_init.append(error)
            iter_x.append(i)
            sys.stdout.write(f'\riteration={i:2d} status={percent:0.2f}% error={error}')
           
    lcurve=np.vstack((iter_x,q_error_pca_init)).T
    np.savetxt('test_10/lcurve'+str(num_iter)+'.txt',lcurve)
```

And now let's plot our learning curves in one plot:
![lcurves](https://user-images.githubusercontent.com/16464387/54758793-22853b80-4bed-11e9-960c-57ad94ca21f4.png)
And here we see this strange behavior: the more we have iterations, the slower is our convergence...
What am I doing wrong?

**Comment by JustGlowing** [comment]
hi @ShrRa , I don't see anything strange with these learning curves. They all converge to the same quantization error (around .3). If you wonder why you converge faster with 10000 iteration, it's because of the learning rate that decreases when the number of iteration increases.

**Comment by ShrRa** [comment]
Ok, sorry, maybe I don't understand the idea, but it means that there is no point in increasing the number of iterations for getting better SOM. If the learning curve has the same shape for different numbers of iterations, how would we determine the optimal number of iterations?

**Comment by JustGlowing** [comment]
Indeed. In your example 10000 iterations are enough for the SOM to converge.

**Comment by ShrRa** [comment]
Ok, thanks.
